### PR TITLE
Feature/activity data classes

### LINKED ISSUE #33: Activity Class
Should represent a single activity object.

Properties

- [x] userID (Num)
- [x] date (String)
- [x] numSteps (Num)
- [x] minutesActive (Num)
- [x] flightsOfStairs (Num)

### LINKED ISSUE #34: ActivityRepository Class
Should hold instances of the Activity class

#### All user methods:

- [x] Avg stairs climbed for a specific date
- [x] Avg steps taken for a  specific date
- [x] Avg minutes active for a specific date

#### Individual User methods:

- [x] For a user, (identified by their userID) how many minutes were they active for a given day (specified by a date)?
- [x] For a user, find their all-time stair climbing record
- [x] For a user, did they reach their step goal for a given day (specified by a date)?
- [x] For a user, find all the days where they exceeded their step goal
- [x] For a user, how many minutes active did they average for a given week (7 days)?
- [x] For a specific day (specified by a date), return the miles a user has walked based on their number of steps (use their strideLength to help calculate this)

### PR DESCRIPTION
**Pull request by Davidschlundtbodien**
# Description
Introduces the Activity and ActivityRepository classes for handling data from api

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor feature (non-breaking change that modifies existing work)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?
- [ ] open index.html
- [x] console.log()
- [ ] dev tools
- [x] npm test

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: test/ActivityRepository-test.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        {"userID": 1, "date": '2019/06/18', "numSteps": 10000, "minutesActive": 10, 'flightsOfStairs': 31},
+        {"userID": 1, "date": '2019/06/19', "numSteps": 11374, "minutesActive": 97, 'flightsOfStairs': 63},
+        {"userID": 1, "date": '2019/06/20', "numSteps": 2643, "minutesActive": 21, 'flightsOfStairs': 23},
+        {"userID": 1, "date": '2019/06/21', "numSteps": 6389, "minutesActive": 40, 'flightsOfStairs': 43},
+        {"userID": 1, "date": '2019/06/22', "numSteps": 11652, "minutesActive": 76, 'flightsOfStairs': 31},
+        {"userID": 2, "date": '2019/06/15', "numSteps": 4230, "minutesActive": 82, 'flightsOfStairs': 12},
+        {"userID": 2, "date": '2019/06/16', "numSteps": 9842, "minutesActive": 50, 'flightsOfStairs': 31},
+        {"userID": 2, "date": '2019/06/17', "numSteps": 5023, "minutesActive": 40, 'flightsOfStairs': 41}
+      ]
+      const userData = [{
```
Comment (joanafbrito) [inline test/ActivityRepository-test.js]: Hey!  Not sure if this is supposed to be a const, it has the same name as the let before the beforeEach statement.

File: src/ActivityRepository.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    const days = reversedDates.slice(dateIndex, dateIndex + 7)
+    const average = days.reduce((currentTotal, day) => {
+      return day.minutesActive + currentTotal
+    }, 0) / days.length
+    return Math.round(average)
+  }
+
+  calculateMilesTraveled(date) {
+    const stepsTaken = this.findUserActivityData(this.currentUser.id, date, "numSteps")
+    const milesResult = (stepsTaken * this.currentUser.strideLength)/5280
```
Comment (joanafbrito) [inline src/ActivityRepository.js:60]: On line 58 there's a string being passed as a argument, just to be cohesive would be better if we user single quotes.

Would you mind commenting here what is the 5280 number means?

**Review by joanafbrito — CHANGES_REQUESTED** [review]
On line 58 there's a string being passed as a argument, just to be cohesive would be better if we user single quotes.

Would you mind commenting here what is the 5280 number means?

Hey! Not sure if this is supposed to be a const, it has the same name as the let before the beforeEach statement.

File: src/ActivityRepository.js (hunk 2 above)
Comment (Davidschlundtbodien) [reply src/ActivityRepository.js:60]: Fixed both! Nice catch.

**Pull request merged**